test/unit/spec/DeviceFingerprint_spec.js

```diff
@@ -17,6 +18,13 @@ function ($, Q, Expect, $sandbox, DeviceFingerprint) {
       window.postMessage(JSON.stringify(message), '*');
     }
 
+    function mockUserAgent(userAgent) {
+      var _navigator = navigator;
+      navigator = new Object();
+      navigator.__proto__ = _navigator;
+      navigator.__defineGetter__('userAgent', function () { return userAgent; });
```

Comment:
I think you probably need an ``afterEach()`` method to restore the ``navigator`` to its original value before it was modified by this test, or else I think subsequent tests after this one will get the the last set navigator value after running this test.